src/OrleansServiceBus/Providers/Streams/EventHub/EventHubPartitionCheckpoint.cs

```diff
@@ -0,0 +1,74 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Microsoft.ServiceBus.Messaging;
+using Orleans.AzureUtils;
+
+namespace Orleans.ServiceBus.Providers
+{
+    internal class EventHubPartitionCheckpoint
+    {
+        private readonly AzureTableDataManager<EventHubPartitionCheckpointEntity> dataManager;
+        private readonly TimeSpan persistCadence;
+
+        private EventHubPartitionCheckpointEntity entity;
+        private Task pendingSave;
+        private DateTime? nextSave;
+
+        public bool Exists { get { return entity != null && entity.Offset != EventHubConsumerGroup.StartOfStream; } }
+
+        public EventHubPartitionCheckpoint(ICheckpointSettings settings, string streamProviderName, string partition)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+            if (string.IsNullOrWhiteSpace(streamProviderName))
+            {
+                throw new ArgumentNullException("streamProviderName");
+            }
+            if (string.IsNullOrWhiteSpace(partition))
+            {
+                throw new ArgumentNullException("partition");
+            }
+            persistCadence = settings.PersistCadence;
+            dataManager = new AzureTableDataManager<EventHubPartitionCheckpointEntity>(settings.TableName, settings.DataConnectionString);
+            entity = EventHubPartitionCheckpointEntity.Create(streamProviderName, settings.DeploymentId, partition);
+        }
+
+        public Task InitTableAsync()
```

Comment:
leaking implementation details... users should not care if this uses azure tables or what